### PR TITLE
Add perPartitionLimit() to QueryBuilder DSL

### LINKED ISSUE #835: Support per partition limits in QueryBuilder
Stargate's `QueryBuilder` does not support `PER PARTITION LIMIT`, even though it is available in CQL.

https://docs.datastax.com/en/cql-oss/3.x/cql/cql_using/useQueryColumnsSort.html

https://docs.datastax.com/en/dse/6.8/cql/cql/cql_using/useQueryColumnsSort.html

This could be useful to Stargate modules in general, but in particular for #828 .

### PR DESCRIPTION
**Pull request by dimas-b**
Closes #835

### CODE REVIEW

File: persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java [hunk 1]
```diff
@@ -665,6 +666,19 @@ public void limit(Integer limit) {
     }
   }
 
+  @DSLAction
+  public void limitPerPartition() {
```
Comment (olim7t) [inline persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java:670]: We should stick to the CQL terminology and name this `perPartitionLimit()`.

Regardless of which name is better, we can't change CQL and I think it's more important to be consistent across the board (for example when people look for the option with autocomplete, etc).

Comment (dimas-b) [reply persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java]: updated

**Review by dougwettlaufer — APPROVED** [review]
Looks like tests need to be re-ran since codacy upload failed.

File: persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java (hunk 1 above)
Comment (dimas-b) [reply persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java]: I renamed field names to follow the `perPartitionLimit` pattern in the latest commit, which also fixes a bug in parameter passing to `Bound`.

**Review by mpenick — APPROVED** [review]

File: persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java [hunk 2]
```diff
@@ -85,7 +85,7 @@
       @SubExpr(
           name = "select",
           definedAs =
-              "select star? column* writeTimeColumn? from (where* limit? orderBy*) allowFiltering?"),
+              "select star? column* writeTimeColumn? from (where* perPartitionLimit? limit? orderBy*) allowFiltering?"),
```
Comment (EricBorczuk) [inline persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java:88]: It's legal to have both a per partition and "overall" limit, right?

**Review by EricBorczuk — APPROVED** [review]
Looks good to me!

File: persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java (hunk 2 above)
Comment (dimas-b) [reply persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java:88]: yes. `PersisteceTest` has a test case for that too.

**Pull request merged**